### ISSUE TITLE
FR: JSON-compliant comments in policy file

### ISSUE THREAD

**Issue by neilschelly** [tailscale/tailscale#12361]
### What are you trying to do?

We parse/process/generate our policy file. It's difficult and/or not possible with a lot of JSON libraries to generate JSON files with the "extra features" like inline comments and trailing comments. Because of this, our policy files are outputted with no comments, making them rather hard to read.

### How should we solve this?

I'd like to request that we can use fields named `__comment__` anywhere in our JSON documents. The fields can just be ignored/stripped from your validator/parser code. This would let us put pure-JSON-compliant comments in our policies.

### What is the impact of not solving this?

Our policy files will continue to be somewhat difficult to read.

**Comment by yqs112358** [comment]
Any consideration to try Json5 as the format of policy file? 🤔

**Comment by neilschelly** [comment]
> Any consideration to try Json5 as the format of policy file? 🤔

No, I was turned off by all the performance problems apparent in the page for the json5 module. I could try it and see what happens though. Our policy is not huge, so the performance concerns may not really matter.

**Comment by neilschelly** [comment]
@yqs112358 I played with json5 a bit just now. It does not preserve comments when it parses a JSON file with comments, so once you want to write out the JSON document for the policy, it will not have any comments anymore.

It does appear that the JSON5 developers would like to implement it, but the issue (https://github.com/dpranke/pyjson5/issues/28) is several years old now.
